src/components/UserInputComponent/BoatMenuComponent.js

```diff
@@ -31,35 +44,231 @@ export default function BoatMenuComponent(props) {
+                }}
+              />
+            </span>
+          </Tooltip>
+        </Grid>
+      </Grid>
+      <Grid paddingBottom={2}>
+        <Typography
+          style={{
+            fontSize: 14,
+            fontWeight: 275,
+            verticalAlign: "middle",
           }}
-          sx={{ width: "99%" }}
-          size="small"
-          renderInput={(params) => (
-            <TextField style={{ backgroundColor: "white" }} {...params} />
-          )}
-        />
-      </Form.Group>
+          color="textSecondary"
+          gutterBottom
+        >
+          Näytä aluksen lisätiedot
+          <Switch
+            size="small"
+            checked={showOld}
+            onChange={(e) => setShowOld(e.target.checked)}
+            inputProps={{ "aria-label": "controlled" }}
+          />
+        </Typography>
+        {showOld ? (
+          <Grid container spacing={1}>
+            {/*Laivan tiedot*/}
+            <Grid item xs={4}>
+              <InputLabel style={{ fontSize: 14 }} id={"Väylän_tunnus"}>
+                Väylän tunnus:
+              </InputLabel>
+              <TextField
+                id="Väylän_tunnus"
+                InputProps={{ sx: { height: 30 } }}
+                value={selectedBoat?.["JNRO"] || ""}
+                disabled
+              />
+            </Grid>
+            <Grid item xs={4}>
+              <InputLabel style={{ fontSize: 14 }} id={"Väylän_nimi"}>
+                Väylän nimi:
+              </InputLabel>
+              <TextField
+                id="Väylän_nimi"
+                InputProps={{ sx: { height: 30 } }}
+                value={selectedBoat?.["VAY_NIMISU"] || ""}
+                disabled
+              />
+            </Grid>
+            <Grid item xs={4}>
+              <InputLabel style={{ fontSize: 14 }} id={"Selite"}>
+                Selite:
+              </InputLabel>
+              <TextField
+                id="Selite"
+                InputProps={{ sx: { height: 30 } }}
+                value={selectedBoat?.["SELITE"] || ""}
+                disabled
+              />
+            </Grid>
+            <Grid item xs={4}>
+              <InputLabel style={{ fontSize: 14 }} id={"Koko"}>
+                Koko:
+              </InputLabel>
+              <TextField
+                id="Koko"
+                InputProps={{ sx: { height: 30 } }}
+                value={selectedBoat?.["KOKO"] || ""}
+                disabled
+              />
+            </Grid>
+            <Grid item xs={4}>
+              <InputLabel style={{ fontSize: 14 }} id={"Runko_kerroin"}>
+                Runko kerroin:
+              </InputLabel>
+              <TextField
+                id="Runko_kerroin"
+                InputProps={{ sx: { height: 30 } }}
+                value={selectedBoat?.["RUNKO_TKERROIN"] || ""}
+                disabled
+              />
+            </Grid>
+          </Grid>
```

Comment:
```suggestion
          <Grid container spacing={1}>
            {/*Laivan tiedot*/}
            {selectedBoatOldFields?.map((oldField) => (
              <Grid key={oldField.id} item xs={4}>
                <InputLabel style={{ fontSize: 14 }} id={oldField.id}>
                  {oldField.id}:
                </InputLabel>
                <TextField
                  id={oldField.id}
                  InputProps={{ sx: { height: 30 } }}
                  value={selectedBoat?.[oldField.field] || ""}
                  disabled
                />
              </Grid>
            ))}
          </Grid>
```
This could be refactored to map an array and create the fields that way. This would require that an array field the old fields ids and fieldValues exist, an example that works with the suggestion would be:

`const selectedBoatOldFields = [
    { id: "Väylän_tunnus", field: "JNRO" },
    { id: "Väylän_nimi", field: "VAY_NIMISU" },
    { id: "Selite", field: "SELITE" },
    { id: "Koko", field: "KOKO" },
    { id: "Runko_kerroin", field: "RUNKO_TKERROIN" },
  ];`

Reply:
yes this would be smart but we still need to name the labels (["VAY_NIMSU"] -> "Vaylän nimi" and ["JNRO"] -> "Vaylän tunnus" and so on + separate the length, beam and draft elements as they are already rendered (yes this could be done  but I call overengineering 😀 )

Reply:
Yeah it might be a bit overengineering, let's leave it like it is :D